### PR TITLE
Fix #3534

### LINKED ISSUE #3534: ofxOsc doesn't make use of boolean OSC 1.1 specifications
The OSC 1.1 specifications are not correctly implemented in ofxOsc and the oscpack extra library.
By now, the following message will not be interpreted:
`/to/my/boolean/path T`
nor this one:
`/to/my/boolean/path F`

By the way, oF developers seems to have choosen the `int`, `float` and `string` replacement like that:
`/to/my/path i 0` or `/to/my/path f 0` or `/to/my/path s true` interpreted as **TRUE** and `/to/my/path i 1` or `/to/my/path f 1` or `/to/my/path s false` interpreted as **FALSE**, but that's not enough and NOT according specifications.

I will propose you some PullReq to make it possible.
Interested?

### PR DESCRIPTION
**Pull request by groolot**

### CODE REVIEW

**Comment by ofTheo** [comment]
Thanks for this PR! 
This looks fairly good to me. 

The only thing that I would be nervous about is changing the values of getTypeName(). 
If people are using that in their code and checking its value then the new values ( which are more correct in terms of the spec ), might cause issues. 

@bakercp @bilderbuchi what do you guys think?

File: addons/ofxOsc/src/ofxOscMessage.cpp [hunk 1]
```diff
@@ -142,6 +154,26 @@ float ofxOscMessage::getArgAsFloat( int index ) const
...
+    if ( getArgType(index) != OFXOSC_TYPE_DOUBLE )
+    {
+        if ( getArgType( index ) == OFXOSC_TYPE_INT32 )
+        {
+            ofLogWarning("ofxOscMessage") << "getArgAsDouble(): converting double to int32 for argument " << index;
+            return ((ofxOscArgInt32*)args[index])->get();
+        }
+        else
+        {
+            ofLogError("ofxOscMessage") << "getArgAsDouble(): argument " << index << " is not a number";
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscMessage.cpp]: what about the other number types (FLOAT, INT64) - those are not accounted for, and would erroneously trip this error.

File: addons/ofxOsc/src/ofxOscMessage.cpp [hunk 2]
```diff
@@ -142,6 +154,26 @@ float ofxOscMessage::getArgAsFloat( int index ) const
 }
 
 
+double ofxOscMessage::getArgAsDouble( int index ) const
+{
+    if ( getArgType(index) != OFXOSC_TYPE_DOUBLE )
+    {
+        if ( getArgType( index ) == OFXOSC_TYPE_INT32 )
+        {
+            ofLogWarning("ofxOscMessage") << "getArgAsDouble(): converting double to int32 for argument " << index;
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscMessage.cpp]: I think there's a mixup here - this function should return a double. If the arg is an int, is should cast the int to double (not the other way round). Also, why warn about it?

File: addons/ofxOsc/src/ofxOscMessage.cpp [hunk 3]
```diff
@@ -237,12 +278,16 @@ ofxOscMessage& ofxOscMessage::copy( const ofxOscMessage& other )
...
-		if ( argType == OFXOSC_TYPE_INT32 )
+		if ( argType == OFXOSC_TYPE_TRUE or argType == OFXOSC_TYPE_FALSE )
+			args.push_back( new ofxOscArgBool( other.getArgAsBool( i ) ) );
+		else if ( argType == OFXOSC_TYPE_INT32 )
 			args.push_back( new ofxOscArgInt32( other.getArgAsInt32( i ) ) );
 		else if ( argType == OFXOSC_TYPE_INT64 )
 			args.push_back( new ofxOscArgInt64( other.getArgAsInt64( i ) ) );
-		else if ( argType == OFXOSC_TYPE_FLOAT )
-			args.push_back( new ofxOscArgFloat( other.getArgAsFloat( i ) ) );
+        else if ( argType == OFXOSC_TYPE_FLOAT )
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscMessage.cpp]: indentation messed up here

File: addons/ofxOsc/src/ofxOscMessage.cpp [hunk 4]
```diff
@@ -237,12 +278,16 @@ ofxOscMessage& ofxOscMessage::copy( const ofxOscMessage& other )
 	for ( int i=0; i<(int)other.args.size(); ++i )
 	{
 		ofxOscArgType argType = other.getArgType( i );
-		if ( argType == OFXOSC_TYPE_INT32 )
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscMessage.cpp]: not really the submitter's fault, but why on earth is this not a switch statement instead of if/else tree? we're testing for enums, after all...

File: addons/ofxOsc/src/ofxOscReceiver.cpp [hunk 5]
```diff
@@ -39,8 +39,12 @@ ofxOscReceiver::ofxOscReceiver()
 	listen_socket = NULL;
 }
 
-void ofxOscReceiver::setup( int listen_port )
+void ofxOscReceiver::setup( int listen_port, bool allowReuse )
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscReceiver.cpp]: where do these unrelated changes come from? they have nothing to do with the PR subject, and that's something that @ofTheo has already merged in, no? I'd be hesitant to touch the reuse code again, tbh.

File: addons/ofxOsc/src/ofxOscSender.cpp [hunk 6]
```diff
@@ -47,12 +47,16 @@ ofxOscSender::~ofxOscSender()
 		shutdown();
 }
 
-void ofxOscSender::setup( std::string hostname, int port )
+void ofxOscSender::setup( std::string hostname, int port, bool enableBroadcast )
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscSender.cpp]: also, the broadcast changes appear here - was there some mixup with the PR branch root?

File: addons/ofxOsc/src/ofxOscSender.cpp [hunk 7]
```diff
@@ -76,7 +80,7 @@ void ofxOscSender::sendBundle( ofxOscBundle& bundle )
 	socket->Send( p.Data(), p.Size() );
 }
 
-void ofxOscSender::sendMessage( ofxOscMessage& message )
+void ofxOscSender::sendMessage( ofxOscMessage& message, bool wrapInBundle )
```
Comment (bilderbuchi) [inline addons/ofxOsc/src/ofxOscSender.cpp]: also, the bundle changes are here - why??

**Comment by bilderbuchi** [comment]
one of your commits - c061f658f562e4deabbb8c3c5f5a733c74065509 - dragged in some unrelated changes. also, it is an apparent imperfect duplicate of 99103c8257926d1baebd1acafb267a588fdb7ba7 - did you rebase your branch or some other fishy git thing?

**Comment by bilderbuchi** [comment]
personally, I would think that `getTypeName` is only a cosmetic function, readable strings for logging or whatever. If people did comparisons, they should use `getType` and compare against the enum values (which is even easier). therefore I think it would be fine to change this. @bakercp do you agree?

**Comment by bilderbuchi** [comment]
I think we should not merge this until the unrelated changes I mentioned above are purged from the PR. After that, and the stuff I commented inline is fixed, it looks good to me.

**Comment by groolot** [comment]
I realise that I made some Git mistakes, and finaly haven't resolved correctly the issue.
I propose you to not take care of that issue and I will resend you some modifications, based on the 0.8.4 branch, without bakercp's ofxOSC-update double's implements.

Do you agree ?

**Comment by bilderbuchi** [comment]
please just base your PR branch off of the current master, that would be best. :-)

**Comment by bilderbuchi** [comment]
the double changes, if corrected in the implementation, would be fine for me, too.
